test/cpp/end2end/xds/xds_override_host_end2end_test.cc

```diff
@@ -267,6 +316,140 @@ TEST_P(OverrideHostTest, DrainingExcludedFromOverrideSet) {
+  // backends
+  CheckRpcSendOk(DEBUG_LOCATION, kNumEchoRpcs);
+  // Cluster with a single backed, all cluster requests end up there
+  EXPECT_THAT(BackendRequestPercentage(backends_[0], kNumEchoRpcs),
+              ::testing::DoubleNear(kWeight1Percent, kErrorTolerance));
+  // Cluster with 3 backends. Backends get distributed there
+  EXPECT_THAT(BackendRequestPercentage(backends_[1], kNumEchoRpcs),
+              ::testing::DoubleNear(kWeight2Percent / 3, kErrorTolerance));
+  EXPECT_THAT(BackendRequestPercentage(backends_[2], kNumEchoRpcs),
+              ::testing::DoubleNear(kWeight2Percent / 3, kErrorTolerance));
+  EXPECT_THAT(BackendRequestPercentage(backends_[3], kNumEchoRpcs),
+              ::testing::DoubleNear(kWeight2Percent / 3, kErrorTolerance));
+  auto session_cookie =
+      GetAffinityCookieHeaderForBackend(DEBUG_LOCATION, 2, 10);
+  ASSERT_FALSE(session_cookie.empty());
+  // All requests go to the backend we requested.
+  CheckRpcSendOk(DEBUG_LOCATION, kNumEchoRpcs,
+                 RpcOptions().set_metadata(session_cookie));
+  EXPECT_EQ(backends_[0]->backend_service()->request_count(), 0);
+  EXPECT_EQ(backends_[1]->backend_service()->request_count(), 0);
+  EXPECT_EQ(backends_[2]->backend_service()->request_count(), kNumEchoRpcs);
+  EXPECT_EQ(backends_[3]->backend_service()->request_count(), 0);
+}
+
+TEST_P(OverrideHostTest, HostGone) {
+  CreateAndStartBackends(4);
+  const char* kNewCluster1Name = "new_cluster_1";
+  const char* kNewEdsService1Name = "new_eds_service_name_1";
+  const char* kNewCluster2Name = "new_cluster_2";
+  const char* kNewEdsService2Name = "new_eds_service_name_2";
+  const uint32_t kWeight1 = std::numeric_limits<uint32_t>::max() / 4;
+  const uint32_t kWeight2 = std::numeric_limits<uint32_t>::max() - kWeight1;
+  const double kErrorTolerance = 0.025;
+  const double kWeight2Percent =
+      static_cast<double>(kWeight2) / std::numeric_limits<uint32_t>::max();
+  const size_t kNumEchoRpcs =
+      ComputeIdealNumRpcs(kWeight2Percent, kErrorTolerance);
+  // Populate EDS and CDS resources.
+  SetEdsCdsResources(kNewCluster1Name, kNewEdsService1Name, 0, 1);
+  SetEdsCdsResources(kNewCluster2Name, kNewEdsService2Name, 1, 4);
+  // Populating Route Configurations for LDS.
+  SetListenerAndRouteConfiguration(
+      balancer_.get(), BuildListenerWithStatefulSessionFilter(),
+      BuildRouteConfiguration(
+          {{kNewCluster1Name, kWeight1}, {kNewCluster2Name, kWeight2}}));
+  WaitForAllBackends(DEBUG_LOCATION, 0, 4);
+  auto session_cookie =
+      GetAffinityCookieHeaderForBackend(DEBUG_LOCATION, 3, 10);
+  ASSERT_FALSE(session_cookie.empty());
+  // Remove backends[1] from cluster 2
+  SetEdsCdsResources(kNewCluster2Name, kNewEdsService2Name, 1, 3);
+  WaitForAllBackends(DEBUG_LOCATION, 0, 3);
+  CheckRpcSendOk(DEBUG_LOCATION, kNumEchoRpcs,
+                 RpcOptions().set_metadata(session_cookie));
+  // Traffic goes to a second cluster, where it is equally distributed between
```

Comment:
I don't think this test needs to care about the exact distribution, since we're not testing round_robin here.  I think that we should just check that the sum of requests seen by the two backends add up to `kNumEchoRpcs`.